lib/MusicBrainz/Server/Controller/WS/js/Edit.pm

```diff
@@ -351,6 +361,19 @@ sub process_artist_credit {
         if defined $data->{artist_credit};
 }
 
+sub process_entered_from {
+    my ($c, $loader, $data) = @_;
+
+    if (defined $data->{enteredFrom}) {
+        my $entered_from = delete $data->{enteredFrom};
+        my $entity_model = type_to_model($entered_from->{entity_type});
+        my $entity = $c->model($entity_model)->get_by_gid($entered_from->{gid});
+        if ($entity) {
+            $data->{entered_from} = $entered_from;
```

Comment:
I'd suggest not submitting the entity name with `enteredFrom`, since it doesn't seem to serve a purpose and can only cause issues (either it's already identical to `$entity->name`, or it contains bad data).
```suggestion
            $data->{entered_from} = {
                entity_type => $entity_type,
                gid => $gid,
                name => $entity->name,
            };
```

Reply:
The point was to have something to display if the release is removed, but I guess your suggestion is to still store the name but just take it from the entity on edit entry? That would seem fine as well and possibly safer, sure.

Reply:
Exactly, we'd store the name but take it from the loaded entity instead.